components/FavoriteButton/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { RiStarSmileLine } from "react-icons/ri";
+import { RiStarSmileFill } from "react-icons/ri";
+import { useContext } from "react";
+import { FavoriteContext } from "../../pages/_app";
+
+export default function FavoriteButton(props) {
+  const favoritecontext = useContext(FavoriteContext);
+  const favoriteButtonHandler = () => {
+    favoritecontext.handleFavorite(props.productcard);
+  };
+
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  const handleFavoriteClick = () => {
+    setIsFavorite(!isFavorite);
+  };
```

Comment:
this function should also be there and global
would pack both into the `_app.js` and then integrate them with use context on both sides.